python/examples/example_errors.ipynb

```diff
@@ -0,0 +1,977 @@
+    "from amici.petab_import import import_petab_problem\n",
+    "from amici.petab_objective import simulate_petab, RDATAS, EDATAS\n",
+    "from amici.plotting import plot_state_trajectories, plot_jacobian\n",
+    "import petab\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "from pathlib import Path\n",
+    "from contextlib import suppress\n",
+    "\n",
+    "\n",
+    "try:\n",
+    "    import benchmark_models_petab\n",
+    "except ModuleNotFoundError:\n",
+    "    # install `benchmark_models_petab` if necessary\n",
+    "    import sys\n",
+    "    !{sys.executable} -m pip install -e git+https://github.com/Benchmarking-Initiative/Benchmark-Models-PEtab.git@master\\#subdirectory=src/python&egg=benchmark_models_petab\n",
+    "    import benchmark_models_petab\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "46500fb0",
+   "metadata": {},
+   "source": [
+    "## Overview\n",
+    "\n",
+    "In the following, we will simulate models contained in the [PEtab Benchmark Collection](https://github.com/Benchmarking-Initiative/Benchmark-Models-PEtab/) to demonstrate a number of simulation failures to analyze and fix them. We use the PEtab format, as it makes model import and simulation much easier, but everything illustrated here, also applies to plain SBML or PySB import.\n",
+    "\n",
+    "If any simulation failures occur, they will be printed via Python logging. \n",
+    "\n",
+    "Programmatically, simulation success can be checked via `ReturnDataView.status`. In case of a successful simulation, and only then, this value corresponds to `amici.AMICI_SUCCESS`.\n",
+    "In case of a simulation error, all quantities in `ReturnData`/`ReturnDataView` will be reported up to the time of failure, the rest will be `NaN`. The likelihood and it's gradient will always be `NaN` in case of failure."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0ad882ac",
+   "metadata": {},
+   "source": [
+    "## `AMICI_TOO_MUCH_WORK` - `mxstep steps taken before reaching tout`\n",
+    "\n",
+    "Let's run a simulation:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ae3cb45e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "petab_problem = benchmark_models_petab.get_problem(\"Fujita_SciSignal2010\")\n",
+    "amici_model = import_petab_problem(petab_problem, verbose=False, force_compile=False)\n",
+    "\n",
+    "np.random.seed(2991)\n",
+    "problem_parameters = dict(\n",
+    "    zip(\n",
+    "        petab_problem.x_free_ids,\n",
+    "        petab_problem.sample_parameter_startpoints(n_starts=1)[0],\n",
+    "    )\n",
+    ")\n",
+    "res = simulate_petab(\n",
+    "    petab_problem=petab_problem, \n",
+    "    amici_model=amici_model,\n",
+    "    problem_parameters=problem_parameters,\n",
+    "    scaled_parameters=True\n",
+    ")\n",
+    "print(\"Status:\", [amici.simulation_status_to_str(rdata.status) for rdata in res[RDATAS]])"
```

Comment:
assert that condition `condition_step_03_0` fails at `t = 3031.8`

Reply:
asserted status codes. 

i think the exact time point may vary slightly on different systems, and is currently not easily accessible. maybe something to add...